File: edge.rb
class Edge
	attr_reader :from_vertex, :to_vertex, :weight
	attr_accessor :effective_weight, :used
	def initialize(from_vertex, to_vertex, weight)
		@from_vertex = from_vertex
		@to_vertex = to_vertex
		@weight = weight
		@effective_weight = weight
		@used = false
		add_self_to_vertices
	end

	def add_self_to_vertices
		@from_vertex.out_edges << self
		@to_vertex.in_edges << self
	end

	def send_bunnies(num_bunnies)
		@from_vertex.current_bunnies -= num_bunnies
		@to_vertex.current_bunnies += num_bunnies
		@effective_weight -= num_bunnies
		@to_vertex.needed_bunnies -= num_bunnies
		@effective_weight = [@from_vertex.current_bunnies, @effective_weight].min
	end
end

File: python_solution/edge.py
class Edge:
	def __init__(self, from_vertex, to_vertex, weight):
		self.from_vertex = from_vertex
		self.to_vertex = to_vertex
		self.weight = weight
		self.effective_weight = weight
		self.used = False
		self.add_self_to_vertices()

	def add_self_to_vertices(self):
		self.from_vertex.out_edges.append(self)
		self.to_vertex.in_edges.append(self)

	def send_bunnies(self, num_bunnies):
		self.from_vertex.current_bunnies -= num_bunnies
		self.to_vertex.current_bunnies += num_bunnies
		self.effective_weight -= num_bunnies
		self.to_vertex.needed_bunnies -= num_bunnies
		self.effective_weight = min([self.from_vertex.current_bunnies, self.effective_weight])
